lib/core/include/irods/data_object_proxy.hpp

```diff
@@ -441,103 +527,68 @@ namespace irods::experimental::data_object
         return duplicate_data_object(*_obj.get());
     } //duplicate_data_object
 
-    /// \brief Finds a replica in the list based on resource hierarchy
+    /// \brief Finds a replica in the list based on resource hierarchy.
     ///
-    /// \param[in] _obj data_object_proxy to search in
+    /// \param[in] _info DataObjInfo to search in
     /// \param[in] _hierarchy resource hierarchy to find
     ///
-    /// \retval replica_proxy if found
-    /// \retval std::nullopt if no replica is found with the provided resource hierarchy
+    /// \retval \p DataObjInfo* Node in linked list if found
+    /// \retval nullptr if no replica is found with the provided hierarchy
     ///
-    /// \since 4.2.9
-    template<typename doi_type>
-    static auto find_replica(data_object_proxy<doi_type>& _obj, std::string_view _hierarchy)
-        -> std::optional<replica::replica_proxy<doi_type>>
+    /// \since 4.3.2
+    static auto find_replica(DataObjInfo& _info, std::string_view _hierarchy) -> DataObjInfo*
     {
-        auto itr = std::find_if(
-            std::begin(_obj.replicas()), std::end(_obj.replicas()),
-            [&_hierarchy](const auto& _r)
-            {
-                return _r.hierarchy() == _hierarchy;
-            });
-
-        if (std::end(_obj.replicas()) != itr) {
-            return *itr;
-        }
-
-        return std::nullopt;
+        namespace ir = irods::experimental::replica;
```

Comment:
Is this used? (same goes for the others)

Reply:
It is not. Relic from a bygone iterative implementation. Will remove